### PR TITLE
Allow changing handle shape of a slider

### LINKED ISSUE #1974: Change slider indicator to a rounded rect
Although this is an opinionated topic, I think for the sake of themeability/skinning, users should be able to modify the look of sliders indicators and choose forms other than `epaint::CircleShape`. For a lack of a better word, the indicator is the small bar (or currently circle) in a slider:
 ```
======0======
       ^
```

A solution could be to use a rectangle and set rounding to `ui.visuals().widgets.inactive.rounding`, although this would change the overall look of egui. Another more backwards compatible solution would be to set the rounding to a default value which produces a circle, and add a method like `rounding()` which would override the rounding behavior.

I have considered building my own widget for this, but there are so many great attributes and methods in the current `Slider` implementation that this would feel overkill to me.

### PR DESCRIPTION
**Pull request by YgorSouza**
* Closes #1974

### CODE REVIEW

File: crates/egui/src/widgets/slider.rs [hunk 1]
```diff
@@ -676,12 +676,36 @@ impl<'a> Slider<'a> {
...
+                        stroke: visuals.fg_stroke,
+                    });
+                }
+                style::HandleShape::Rect { aspect_ratio } => {
+                    let v = match self.orientation {
+                        SliderOrientation::Horizontal => Vec2::new(radius, radius / aspect_ratio),
+                        SliderOrientation::Vertical => Vec2::new(radius / aspect_ratio, radius),
+                    };
+                    let v = v + Vec2::splat(visuals.expansion);
+                    let rect = Rect {
```
Comment (emilk) [inline crates/egui/src/widgets/slider.rs:695]: You could use `Rect::from_center_size` here

File: crates/egui/src/widgets/slider.rs [hunk 2]
```diff
@@ -720,12 +744,16 @@ impl<'a> Slider<'a> {
...
         let limit = match self.orientation {
             SliderOrientation::Horizontal => rect.height(),
             SliderOrientation::Vertical => rect.width(),
         };
-        limit / 2.5
+        let aspect_ratio = match handle_shape {
+            style::HandleShape::Circle => 1.0,
+            style::HandleShape::Rect { aspect_ratio } => *aspect_ratio,
+        };
+        limit / (2.5 / aspect_ratio)
```
Comment (emilk) [inline crates/egui/src/widgets/slider.rs:756]: If we define the radius to be on the cross axis (i.e. the Y axis on a horizontal slider), then we don't need to use `aspect_ratio` here

**Review by emilk — COMMENTED** [review]
Looks good, but adding the settings to every `WidgetVisuals` seems overkill - I think the belongs in `Visuals` instead

**Comment by YgorSouza** [comment]
> Looks good, but adding the settings to every `WidgetVisuals` seems overkill - I think the belongs in `Visuals` instead

I thought about that, but I figured maybe someone could want to change the aspect ratio or shape when the slider is hovered or clicked, as an alternative to the `expansion` setting, and that would be very hard to do without the setting on `WidgetVisuals`. On the other hand, it does mean that everyone who doesn't want that is going to have to change this setting 3 times instead of 1, so maybe it's not such a great tradeoff. I'll change it.

File: crates/egui/src/style.rs [hunk 3]
```diff
@@ -1877,3 +1900,22 @@ fn rounding_ui(ui: &mut Ui, rounding: &mut Rounding) {
...
     });
 }
+
+fn handle_shape_ui(ui: &mut Ui, handle_shape: &mut HandleShape) {
+    ui.label("Widget handle shape");
+    ui.horizontal(|ui| {
+        ui.radio_value(handle_shape, HandleShape::Circle, "Circle");
+        if ui
+            .radio(
+                matches!(handle_shape, HandleShape::Rect { aspect_ratio: _ }),
```
Comment (emilk) [inline crates/egui/src/style.rs:1910]: nit

```suggestion
                matches!(handle_shape, HandleShape::Rect { .. }),
```

File: crates/egui/src/widgets/slider.rs [hunk 4]
```diff
@@ -284,6 +286,15 @@ impl<'a> Slider<'a> {
         self
     }
 
+    /// Change the shape of the slider handle
+    ///
+    /// This setting can be enabled globally for all sliders with [`Visuals::handle_shape`].
+    /// Changing it here will override the above setting ONLY for this individual slider.
+    pub fn handle_shape(mut self, handle_shape: HandleShape) -> Self {
```
Comment (emilk) [inline crates/egui/src/widgets/slider.rs:293]:
```suggestion
    #[inline]
    pub fn handle_shape(mut self, handle_shape: HandleShape) -> Self {
```

File: crates/egui_demo_lib/src/demo/sliders.rs [hunk 5]
```diff
@@ -132,23 +137,43 @@ impl super::View for Sliders {
...
+                    matches!(handle_shape, HandleShape::Rect { aspect_ratio: _ }),
+                    "Rectangle",
+                )
+                .clicked()
+            {
+                *handle_shape = HandleShape::Rect { aspect_ratio: 0.5 };
+            }
+            if let HandleShape::Rect { aspect_ratio } = handle_shape {
+                ui.add(Slider::new(aspect_ratio, 0.1..=3.0).text("Aspect ratio"));
+            }
```
Comment (emilk) [inline crates/egui_demo_lib/src/demo/sliders.rs:173]: Duplicated code - I suggest you create an `impl HandleSape { fn ui(&mut self, ui: &mut Ui) { … } }` for this in `style.rs`

**Review by emilk — APPROVED** [review]

**Pull request merged**